main.cpp

```diff
@@ -56,15 +56,18 @@ Purpose:  This project will show you the difference between member functions and
 #include <string>
 struct T
 {
-    T(<#type name#> v, const char* <#variable name#>)   //1
-    //2
-    //3
+    T(int v, const char* n) : value(v),
+                              name(n) {}           //1
+    int value; //2
+    std::string name; //3
 };
 
-struct <#structName1#>                                //4
+struct Comparison                                //4
 {
-    <#type name#> compare(<#type name#> a, <#type name#> b) //5
+    T* compare(T* a, T* b) //5
     {
+        if (a == nullptr || b == nullptr)
+            return nullptr;
         if( a->value < b->value ) return a;
         if( a->value > b->value ) return b;
         return nullptr;
```

Comment:
Now you have 2 `return nullptr;` statements
Revise your logic to have only 1 `return nullptr;` statement in this entire function.
- [ ] revise